### PR TITLE
feat: update ci

### PR DESCRIPTION
**Pull request by greged93**
Time spent on this PR: 0.5 day

Resolves: #849 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Improve the CI for the rpc binary generation.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/workflows/kakarot_rpc.yml [hunk 1]
```diff
@@ -1,59 +1,97 @@
...
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Setup the Kakarot submodule
+        run: make setup
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: mkdir -p ./artifacts/contracts && mv ./lib/kakarot/build/* ./artifacts/contracts
```
Comment (Eikix) [inline .github/workflows/kakarot_rpc.yml:49]: it's this redundant with extract_abi.sh script

File: .github/workflows/kakarot_rpc.yml [hunk 2]
```diff
@@ -1,59 +1,97 @@
...
+      - run: mkdir ./artifacts && mv /tmp/artifacts/abi/* ./artifacts
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+      - uses: taiki-e/install-action@cross
+      - name: Log in to Docker
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io --username ${DOCKER_USERNAME} --password-stdin
```
Comment (Eikix) [inline .github/workflows/kakarot_rpc.yml:86]: why do we need to manually connect to docker? i thought the action was doing it

File: .github/workflows/kakarot_rpc.yml (hunk 1 above)
Comment (greged93) [reply .github/workflows/kakarot_rpc.yml]: yes it was, removed

File: .github/workflows/kakarot_rpc.yml (hunk 2 above)
Comment (greged93) [reply .github/workflows/kakarot_rpc.yml:81]: So in order to install `tonistiigi/binfmt`, which is a cross platform emulator, I need to run docker as privileged, and for this I need to be logged in.

**Review by Eikix — APPROVED** [review]
lgtm

**Review by Eikix — CHANGES_REQUESTED** [review]
We could add the makefile change?

**Review by Eikix — APPROVED** [review]

**Pull request merged**